resources/czi_affiliation_links_pipeline/pipeline.py

```diff
@@ -0,0 +1,294 @@
+import argparse
+import boto3
+import csv
+import re
+import requests
+import xml.etree.ElementTree as ET
+
+from botocore import UNSIGNED
+from botocore.config import Config
+from collections import defaultdict
+from multiprocessing.pool import Pool
+
+
+s3_client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+
+PMC_BUCKET = "pmc-oa-opendata"
+OPENALEX_URL = "https://api.openalex.org"
+
+
+def extract_citation_number(mention):
+    """
+    Extract the number of the formal citation from the mention, if the formal
+    citation exists. For example, from the mention "We used Apache Spark [12]
+    to process the logs.", the citation number "12" should be extracted.
+
+    :param mention: Software mention object
+    :return: Formal citation number or None
+    """
+
+    regex = re.escape(mention["software"]) + " *\[([1-9]\d*)"
```

Comment:
You sometimes see citations like `[12, 15]`, but those are kind of an edge case. If you agree this might be worth thinking about later, maybe we can open an issue to come back to it?

Reply:
Will do, thanks!